casper/src/main/scala/coop/rchain/casper/engine/Engine.scala

```diff
@@ -118,6 +117,10 @@ object Engine {
         init,
         disableStateExporter
       )
+      // this is required because before https://github.com/rchain/rchain/pull/3459 genesis as not added to
+      // last finalized storage on network init.
+      _ <- BlockDagStorage[F].recordDirectlyFinalised(approvedBlock.candidate.block.blockHash)
+      _ <- BlockDagStorage[F].addFinalizedBlockHashes(List(approvedBlock.candidate.block.blockHash))
```

Comment:
Can we do this as part of migration?

Reply:
We don't have approved block when migration happens - it is done even before blockdag storage is created, which is required for Casper when approved block is available.